src/handlers/VehicleTestController.ts

```diff
@@ -1,58 +1,230 @@
+    filters: models.ITestResultFilters
+  ): Promise<models.ITestResult[]> {
+    if (
+      !filters.systemNumber ||
+      !utils.ValidationUtil.validateGetTestResultFilters(filters)
+    ) {
+      throw new models.HTTPError(400, enums.MESSAGES.BAD_REQUEST);
+    }
     const result = await this.dataProvider.getTestResultBySystemNumber(filters);
     if (result.length === 0) {
-       throw new models.HTTPError(404, enums.ERRORS.NoResourceMatch);
-     }
+      throw new models.HTTPError(404, enums.ERRORS.NoResourceMatch);
+    }
     return result;
   }
 
   /**
    * To fetch test results by Tester Staff Id
    * @param filters test results filters for search
    */
-  public async getTestResultByTestStaffId(filters: models.ITestResultFilters): Promise<models.ITestResult[]> {
-    if (!filters.testerStaffId || !ValidationUtil.validateGetTestResultFilters(filters)) {
+  public async getTestResultByTestStaffId(
+    filters: models.ITestResultFilters
+  ): Promise<models.ITestResult[]> {
+    if (
+      !filters.testerStaffId ||
+      !utils.ValidationUtil.validateGetTestResultFilters(filters)
+    ) {
       throw new models.HTTPError(400, enums.MESSAGES.BAD_REQUEST);
-      }
-    const result = await this.dataProvider.getTestResultByTesterStaffId(filters);
+    }
+    const result = await this.dataProvider.getTestResultByTesterStaffId(
+      filters
+    );
     console.log("testResults inside controller:", result);
     if (result.length === 0) {
       throw new models.HTTPError(404, enums.ERRORS.NoResourceMatch);
-       }
+    }
     return result;
   }
 
   /**
    * A factory method used to fetch a strategy for calculating expiry date based on a strategy mapping JSON file.
    * @param testTypeForExpiry an input object which is used to calculate expiry based on test type.
    */
-  public getExpiryStrategy(testTestForExpiry: TestTypeForExpiry): IExpiryDateStrategy {
-    const selectedStrategy = ExpiryDateStrategyFactory.GetExpiryStrategy(testTestForExpiry, this.dateProvider);
+  public getExpiryStrategy(
+    testTestForExpiry: TestTypeForExpiry
+  ): IExpiryDateStrategy {
+    const selectedStrategy = ExpiryDateStrategyFactory.GetExpiryStrategy(
+      testTestForExpiry,
+      this.dateProvider
+    );
     return selectedStrategy;
   }
+
+  public async insertTestResult(payload: models.ITestResultPayload) {
+    try {
+      utils.ValidationUtil.validateInsertTestResultPayload(payload);
+      console.info("validation is success");
+      payload = utils.MappingUtil.setCreatedAtAndLastUpdatedAtDates(payload);
+      const testTypeParams: models.TestTypeParams = {
+        vehicleType: payload.vehicleType,
+        vehicleSize: payload.vehicleSize,
+        vehicleConfiguration: payload.vehicleConfiguration,
+        vehicleAxles: payload.noOfAxles,
+        euVehicleCategory: payload.euVehicleCategory,
+        vehicleClass: payload.vehicleClass?.code,
+        vehicleSubclass: payload.vehicleSubclass?.[0],
```

Comment:
Will it always be the first element of the array if present? I was wondering if we needed to get the value by key

Reply:
This has been moved as is from the original logic on develop.